Assets/SuperMultiplayerShooter/Scripts/PNManager.cs

```diff
@@ -117,6 +124,53 @@ public async Task<bool> GetUserMetadata(string Uuid)
             }
             return true;
         }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Update the User Metadata given the UserId.
+    /// </summary>
+    /// <param name="Uuid">UserId of the Player</param>
+    public async Task<bool> UpdateUserMetadata(string Uuid, Dictionary<string, object> metadata)
+    {      
+        //If they do not exist, pull in their metadata (since they would have already registered when first opening app), and add to cached players.                
+        // Get Metadata for a specific UUID
+        PNResult<PNGetUuidMetadataResult> getUuidMetadataResponse = await pubnub.GetUuidMetadata()
+            .Uuid(Uuid)
+            .IncludeCustom(true)
+            .ExecuteAsync();
+        PNGetUuidMetadataResult getUuidMetadataResult = getUuidMetadataResponse.Result;
+        PNStatus status = getUuidMetadataResponse.Status;
+        if (!status.Error && getUuidMetadataResult != null)
+        {
+            UserMetadata meta = new UserMetadata
```

Comment:
You are not using the metadata parameter are you?  I think you said this functionality was still to do but be careful here not to wipe out the player hats which are also stored in the custom meta data